deeplearning/ml4pl/models/ggnn/ggnn_modules.py

```diff
@@ -41,67 +42,102 @@ class GGNNModel(nn.Module):
   def __init__(self, config, pretrained_embeddings=None, test_only=False):
     super().__init__()
     self.config = config
+
+    # input layer
     self.node_embeddings = NodeEmbeddings(config, pretrained_embeddings)
+
     self.ggnn = GGNNProper(config)
-    self.nodewise_readout = NodewiseReadout(config)
+
+    # maybe tack on the aux readout
+    self.has_aux_input = getattr(self.config, "has_aux_input", False)
+
+    if self.has_aux_input:
+      self.aux_readout = AuxiliaryReadout(config)
+
     # make readout available to label_convergence tests in GGNN Proper (at runtime)
     if self.config.unroll_strategy == "label_convergence":
-      self.ggnn.nodewise_readout = self.nodewise_readout
-
-    self.graphlevel_readout = None
-    if config.has_graph_labels:
-      self.graphlevel_readout = AuxiliaryReadout(config)
+      assert (
+        not self.config.has_aux_input
+      ), "aux_input is not supported with label_convergence"
 
     # eval and training
     self.loss = Loss(config)
     self.metrics = Metrics()
 
     # not instantiating the optimizer should save 2 x #model_params of GPU memory, bc. Adam
     # carries two momentum params per trainable model parameter.
+
+    # move model to device before making optimizer!
+    self.dev = (
+      torch.device("cuda")
+      if gpu_scheduler.LockExclusiveProcessGpuAccess()
+      else torch.device("cpu")
+    )
+
+    self.to(self.dev)
```

Comment:
here it moves ...

Reply:
Oh I see now. Done 694b7b7268d18fc5811b9b121cc3e443e890a627